test/Integration/ApplicationBootstrapTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-mvc-middleware for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-mvc-middleware/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Mvc\Middleware\Integration;
+
+use PHPUnit\Framework\TestCase;
+use Zend\Mvc\Middleware\MiddlewareListener;
+use Zend\Mvc\MiddlewareListener as DeprecatedMiddlewareListener;
+
+/**
+ * @group integration
+ * @coversNothing
+ */
+class ApplicationBootstrapTest extends TestCase
+{
+    use ApplicationTrait;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->setUpApplication();
+    }
+
+    protected function tearDown()
+    {
+        parent::tearDown();
+        $this->tearDownApplication();
```

Comment:
I think we should swap these statements here:
first tearDownApplication and then teatDown the test case.